### PR TITLE
[EM-7092] Updated file retrieve to populate stored_in field

### PR DESCRIPTION
**Pull request by gammaseeker**
### Description
In order to fix the redaction bug on classic, the file retrieval needs to also grab the stored_in value when RemoteFiles does it retrieval.

### References
* Jira: https://zendesk.atlassian.net/browse/EM-7092

### Risks
* Low: change is gated behind an option that acts as an arturo.

### CODE REVIEW

File: lib/remote_files/file.rb [hunk 1]
```diff
@@ -79,6 +79,9 @@ def retrieve!
           next unless file
           @content      = file.content
           @content_type = file.content_type
+          if @options[:populate_stored_in]
```
Comment (rmartz00) [inline lib/remote_files/file.rb:82]: This won't raise a KeyError or anything if that value isn't set in the `@options`? If it does then may need to add a check for whether the key exists as well.

Comment (rmartz00) [reply lib/remote_files/file.rb:82]: Also need to add the new option to the list of known keys [here](https://github.com/zendesk/remote_files/pull/26/files#diff-eb648593d3efb23402e60618788bbf355b778db93d29db25762761ec5c743c4fR6)

**Review by rmartz00 — CHANGES_REQUESTED** [review]
Please also test against this in 998 before merging this if possible. Thanks!

File: test/resque_job_test.rb [hunk 2]
```diff
@@ -21,6 +21,7 @@
         :stored_in     => [:s3],
         :foo           => :bar,
         :configuration => :default,
+        :populate_stored_in => nil,
```
Comment (dpumford) [inline test/resque_job_test.rb:24]: Might want to have a test for when this is `true`.

**Review by rodrigoalvesvieira — APPROVED** [review]

File: test/file_test.rb [hunk 3]
```diff
@@ -135,25 +135,28 @@
 
   describe '#retrieve!' do
     before do
-      @file_with_content = RemoteFiles::File.new('identifier', :content => 'content', :content_type => 'content_type')
+      @file_with_content = RemoteFiles::File.new('identifier', :content => 'content', :content_type => 'content_type', :populate_stored_in => true)
```
Comment (rmartz00) [inline test/file_test.rb:138]: Should there also be a test where `populate_stored_in` is `nil` since that seems like the default behavior?

Comment (gammaseeker) [reply test/file_test.rb]: I updated the unit tests to use `populate_stored_in` when it is `nil` and another section for when it is `true`

File: test/file_test.rb [hunk 4]
```diff
@@ -135,10 +135,11 @@
 
   describe '#retrieve!' do
     before do
-      @file_with_content = RemoteFiles::File.new('identifier', :content => 'content', :content_type => 'content_type')
+      @file_with_content = RemoteFiles::File.new('identifier', :content => 'content', :content_type => 'content_type', :populate_stored_in => nil)
```
Comment (rmartz00) [inline test/file_test.rb:138]: There should be an additional test with `populate_stored_in` where it checks that `stored_in` does not get set, somewhere [here](https://github.com/zendesk/remote_files/pull/26/files#diff-620c2f7c8c17aca1391ba33ec06310fc68e8470f4a75260a40e6c28a462da446R159)

File: lib/remote_files/file.rb [hunk 5]
```diff
@@ -79,6 +81,8 @@ def retrieve!
           next unless file
           @content      = file.content
           @content_type = file.content_type
+          # :populate_stored_in is a boolean
+          @stored_in = file.stored_in if options.key?(:populate_stored_in) && (@options[:populate_stored_in])
```
Comment (rmartz00) [inline lib/remote_files/file.rb:85]: Not sure if this changed or if I just noticed but why do we access `@options[:populate_stored_in]` both here and [here](https://github.com/zendesk/remote_files/pull/26/files#diff-eb648593d3efb23402e60618788bbf355b778db93d29db25762761ec5c743c4fR17)? Do we not need the instance var or is it better to just check `key?` there as well and then store it to use here?

File: test/file_test.rb [hunk 6]
```diff
@@ -155,6 +155,14 @@
         @file.content.must_equal 'content'
         @file.content_type.must_equal 'content_type'
       end
+
+      it 'verifies stored_in does not get populated' do
```
Comment (rmartz00) [inline test/file_test.rb:159]: nit: better description here might be "does not fill in stored_in"

File: lib/remote_files/file.rb (hunk 5 above)
Comment (gammaseeker) [reply lib/remote_files/file.rb:85]: That if condition was written before I implemented the instance var, I forgot to update that if condition after those changes.

**Comment by sponee** [comment]
@gaecro-zendesk heads up this is part 2/2 of the comment redaction logic fix.

**Review by rmartz00 — APPROVED** [review]
